MultiThreading.Task5.Threads.SharedCollection/Program.cs

```diff
@@ -5,11 +5,21 @@
  * Use Thread, ThreadPool or Task classes for thread creation and any kind of synchronization constructions.
  */
 using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace MultiThreading.Task5.Threads.SharedCollection;
 
 class Program
 {
+    private const int CollectionLength = 10;
+    private static readonly ManualResetEventSlim _producerAddedNewValue = new(false);
```

Comment:
It seems like a single ManualResetEventSlim would be enough to synchronize 22 threads, isn't it?